src/lib.rs

```diff
@@ -32,6 +35,36 @@ use zenoh::{
 };
 
 const UATTRIBUTE_VERSION: u8 = 1;
+const THREAD_NUM: usize = 10;
+
+type CallbackChannelMessage = (Arc<dyn UListener>, Result<UMessage, UStatus>);
+
+struct CallbackThreadPool {
```

Comment:
@evshary, @sophokles73, @AnotherDaniel --

There's a [discussion](https://github.com/eclipse-uprotocol/up-cpp/pull/99#discussion_r1580105022) going on over in cpp-land during their API redesign on whether a common utility like a thread pool implementation should be offered by `up-cpp` so that all `up-client-foo-cpp` can benefit.

What do we think about offering the same for `up-rust`?

Frankly, I'm kinda not in favor of offering the same in `up-rust` because setting up a thread pool is _much_ easier in Rust than in C++. (~40 lines of code in Rust using very standard betteries-included crates, ~200 lines of code in C++)

I do think we should offer guidance within `up-rust` to implementors of `UTransport` to avoid blocking on send at the very minimum and improve the example at the bottom of `utransport.rs` to include use of a thread-pool.

Curious to hear what y'all think though.

Reply:
I second it. It's not too difficult to do the same thing in Rust. We just need to let developers know they need to avoid blocking callbacks.